corehq/apps/accounting/tests/test_stripe_payment_method_with_api.py

```diff
@@ -0,0 +1,223 @@
+        self.assertTrue(self.billing_account.auto_pay_enabled)
+
+        # Use the same card for the second billing account
+        self.payment_method.set_autopay(self.card, billing_account_2, None)
+        self.assertEqual(self.card.metadata, {"auto_pay_{}".format(self.billing_account.id): 'True',
+                                            "auto_pay_{}".format(billing_account_2.id): 'True'})
+
+    def test_unset_autopay(self):
+        self.payment_method.set_autopay(self.card, self.billing_account, None)
+        self.assertEqual(self.card.metadata, {"auto_pay_{}".format(self.billing_account.id): 'True'})
+
+        self.payment_method.unset_autopay(self.card, self.billing_account)
+
+        self.assertEqual(self.card.metadata, {"auto_pay_{}".format(self.billing_account.id): 'False'})
+        self.assertIsNone(self.billing_account.auto_pay_user)
+        self.assertFalse(self.billing_account.auto_pay_enabled)
+
+    def test_get_stripe_customer_if_existed(self):
+        customer = self.payment_method._get_or_create_stripe_customer()
+        self.assertEqual(customer.id, self.stripe_customer.id)
+
+    def test_create_stripe_customer_if_not_existed(self):
+        web_user_email = generator.create_arbitrary_web_user_name()
+        payment_method = StripePaymentMethod(web_user=web_user_email)
+        customer = payment_method._get_or_create_stripe_customer()
+        self.assertEqual(customer.email, web_user_email)
+        self.addCleanup(customer.delete)
+
+    def test_all_cards_raise_authentication_error_when_stripe_key_is_wrong(self):
+        stripe.api_key = "12345678"
+        with self.assertRaises(stripe.error.AuthenticationError):
+            self.payment_method.all_cards
+
+    def test_all_cards_return_the_correct_collection_of_cards_for_a_customer(self):
+        # Get the payment methods that is associated with the customer
+        payment_methods = stripe.PaymentMethod.list(
+            customer=self.stripe_customer.id,
+            type="card",
+        )
+        cards = self.payment_method.all_cards
+        actual_card_ids = [card.fingerprint for card in cards]
+        expected_card_ids = [payment_method.card.fingerprint for payment_method in payment_methods]
+        self.assertCountEqual(actual_card_ids, expected_card_ids)
+
+    def test_all_cards_return_empty_array_for_customer_have_no_cards(self):
+        payment_method = StripePaymentMethod(web_user="no_card@gmail.com")
+        payment_method.save()
+        cards = payment_method.all_cards
+        self.assertEqual(len(cards), 0)
+
+    def test_all_cards_return_empty_array_if_no_stripe_key(self):
+        stripe.api_key = None
+        original_key = settings.STRIPE_PRIVATE_KEY
+        settings.STRIPE_PRIVATE_KEY = None
+        try:
+            self.assertEqual(len(self.payment_method.all_cards), 0)
+        finally:
+            settings.STRIPE_PRIVATE_KEY = original_key
```

Comment:
No need to change, but I thought of a cleaner way to do this
```suggestion
        from unittest.mock import patch
        with patch.object(settings, "STRIPE_PRIVATE_KEY", None):
            self.assertEqual(len(self.payment_method.all_cards), 0)
```
The import could be moved to the top of the file.

Reply:
Love this!!!!! Thank you!